vortex-datafusion/src/persistent/config.rs

```diff
@@ -33,13 +36,15 @@ impl VortexFile {
 pub struct VortexTableOptions {
     pub(crate) data_files: Vec<VortexFile>,
     pub(crate) schema: Option<SchemaRef>,
+    pub(crate) ctx: Arc<Context>,
 }
 
 impl VortexTableOptions {
-    pub fn new(schema: SchemaRef, data_files: Vec<VortexFile>) -> Self {
+    pub fn new(schema: SchemaRef, data_files: Vec<VortexFile>, ctx: Arc<Context>) -> Self {
```

Comment:
99% of the time (100% of the time right now) you're gonna just be using the default codecs.

maybe it'd make more sense to have an `extra_encodings` config rather than requiring clients to build a whole context just to pass defaults

Reply:
I think I agree with you. I am not happy with the flow here. I think there's probably just a default that has to have erverything for now.